webcomponents/tool/src/main/frontend/packages/sakai-lti-popup/src/SakaiLTIPopup.js

```diff
@@ -29,21 +30,22 @@ export class SakaiLTIPopup extends SakaiElement {
   }
 
   launchPopup() {
-
     window.open(this.launchUrl, "_blank");
     document.getElementById("sakai-lti-popup-" + this.randomId).style.display = "none";
     document.getElementById("sakai-lti-popup-hidden-" + this.randomId).style.display = "block";
     return false;
   }
 
-  shouldUpdate() {
-    return this.preLaunchText && this.postLaunchText && this.launchUrl;
+  shouldUpdate(changedProperties) {
+    return changedProperties.has("preLaunchText") || changedProperties.has("postLaunchText") ||
+          changedProperties.has("launchUrl") || changedProperties.has("auto");
```

Comment:
You should not need to do any of this because these properties are already reactive. All you are doing here is repeating lit's internal logic when any one of these four reactive properties is changed.